### PR TITLE
Add configuration for two ElasticSearch clusters

### PR DESCRIPTION
**Pull request by tarrow**
This PR adds support for configuring two ElasticSearch clusters using environment variables.
It is required to have at least one cluster but it is possible to have an optional second cluster that
is written to but does is not used to serve search results to users.

To clearly differentiate between the types of cluster it uses the same terminology as CirrusSearch
where the default cluster is the one used for serving content to users.

We hope to use these configuration settings to move to a new version of elasticsearch so these new configurations
for elasticsearch hosts do not automatically load CirrusSearch's ES6 compatibility layer but allow it to be optionally enabled.

The original ENV vars for configuring the default ElasticSearch cluster remain present and keep their current behaviour: it always loads the ES6 compatibility later. These are marked as deprecated and should be removed in future.

The PR is derived from https://github.com/wbstack/mediawiki/pull/354 but deliberately excludes additional adjustments to the backend action api parameters that makes these changes coupled to changes in the platform api.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,8 +42,15 @@ This part describes the Docker image that this repo is designed to build. People
 - `MW_RECAPTCHA_SITEKEY`
 - `MW_RECAPTCHA_SECRETKEY`
 - `PLATFORM_API_BACKEND_HOST`: points to an internal mode wbstack api service
-- `MW_ELASTICSEARCH_HOST`: elasticsearch hostname
-- `MW_ELASTICSEARCH_PORT`: elasticsearch port
+- `MW_ELASTICSEARCH_HOST`: DEPRECATED: elasticsearch hostname. If set "default" settings below are ignored.
+- `MW__ELASTICSEARCH_PORT`: DEPRECATED: elasticsearch hostname
```
Comment (rosalieper) [inline README.md:46]: ES port

File: README.md [hunk 2]
```diff
@@ -42,8 +42,15 @@ This part describes the Docker image that this repo is designed to build. People
...
+- `MW_DEFAULT_ELASTICSEARCH_HOST`: default (i.e. where data is served to users from) elasticsearch hostname
+- `MW_DEFAULT_ELASTICSEARCH_PORT`: default elasticsearch port
+- `MW_DEFAULT_ELASTICSEARCH_ES6`: use ES6 compatability layer with default cluster
```
Comment (rosalieper) [inline README.md:49]: maybe we should describe it in the way that it is obvious it is a boolean.

**Review by rosalieper — COMMENTED** [review]
looking at the code and all looks great apart from a minor typo. I just need to test it now.

File: README.md (hunk 1 above)
Comment (tarrow) [reply README.md:46]: ah! yes! Hoist by my own copy paste petard

File: README.md (hunk 2 above)
Comment (tarrow) [reply README.md]: Done!

File: dist-persist/wbstack/src/Settings/LocalSettings.php [hunk 3]
```diff
@@ -647,19 +647,45 @@ function onSkinTemplateNavigationUniversal( SkinTemplate $skin, array &$links )
...
-                    'type' => \CirrusSearch\Elastica\ES6CompatTransportWrapper::class,
-                    'wrapped_transport' => 'Http'
+
+    function getElasticClusterConfig( string $prefix ) {
+        $config = [
+            'host' => getenv( $prefix . 'HOST' ),
+            'port' => getenv( $prefix . 'PORT' )
+        ];
+
+        if ( getEnv( $prefix . 'ES6' ) ) {
```
Comment (m90) [inline dist-persist/wbstack/src/Settings/LocalSettings.php:657]: Very nitpicky, but this does not work as described in the README (i.e. it does not check if it's `"true"`).

File: README.md [hunk 4]
```diff
@@ -42,8 +42,15 @@ This part describes the Docker image that this repo is designed to build. People
 - `MW_RECAPTCHA_SITEKEY`
 - `MW_RECAPTCHA_SECRETKEY`
 - `PLATFORM_API_BACKEND_HOST`: points to an internal mode wbstack api service
-- `MW_ELASTICSEARCH_HOST`: elasticsearch hostname
-- `MW_ELASTICSEARCH_PORT`: elasticsearch port
+- `MW_ELASTICSEARCH_HOST`: DEPRECATED: elasticsearch hostname. If set "default" settings below are ignored.
```
Comment (m90) [inline README.md:45]: Question about marking this as `DEPRECATED`. Considering we have sailed through a super smooth migration and now want to be using ES7 only. Would we not want to go back to using this setting?

**Review by m90 — COMMENTED** [review]
I left some comments around naming, but nothing major. The logic for looking up the config seems solid.

I just wanted to mention that there would still be the option of making the env vars comma separated lists of multiple hosts. This would mean we'd have way less changes in the charts, but I also see this will be confusing to configure, so maybe let's not do it.

File: README.md [hunk 5]
```diff
@@ -42,8 +42,15 @@ This part describes the Docker image that this repo is designed to build. People
...
+- `MW_ELASTICSEARCH_PORT`: DEPRECATED: elasticsearch port
+- `MW_DEFAULT_ELASTICSEARCH_HOST`: default (i.e. where data is served to users from) elasticsearch hostname
```
[the remark this answers is not in the document]
Comment (tarrow) [reply README.md:47]: yeah, you've got the logic exactly right. Cirrus internally insists on calling these default so sticking with that name made sense to me. I'd been keen to move away from secondary and wasted a few minutes thinking about it myself but finding a good name eluded me.

`WRITE_ONLY` was the best alternative I could come up with. What do you think about that?

File: README.md [hunk 6]
```diff
@@ -42,8 +42,15 @@ This part describes the Docker image that this repo is designed to build. People
...
+- `MW_DEFAULT_ELASTICSEARCH_PORT`: default elasticsearch port
+- `MW_DEFAULT_ELASTICSEARCH_ES6`: use ES6 compatability layer with default cluster. Set to "true" or leave unset
+- `MW_SECONDARY_ELASTICSEARCH_ENABLED`: is secondary (i.e. where data is concurrently written but not served to users) elasticsearch enabled. Set to "true" or leave unset
```
[the remark this answers is not in the document]
Comment (tarrow) [reply README.md:50]: yeah, we win very little (nothing except the opportunity to have a host stored that isn't used). I'll strip it out

File: README.md (hunk 4 above)
Comment (tarrow) [reply README.md:45]: I was thinking that we would want to leave the capability to have a second write only cluster there.
For example, if we want to migrate to a second 7 cluster in the future with different field mappings

**Comment by m90** [comment]
I created a chart PR here https://github.com/wbstack/charts/pull/122

**Review by m90 — APPROVED** [review]

**Pull request merged**